### PR TITLE
Make the log file name configurable and include timestamp by default

### PR DESCRIPTION
**Pull request by ANogin**
- [X] I have reviewed the [OFRAK contributor guide](https://ofrak.com/docs/contributor-guide/getting-started.html) and attest that this pull request is in accordance with it.

**One sentence summary of this PR (This should go in the CHANGELOG!)**
By default, the ofrak log is now `ofrak-YYYYMMDDhhmmss.log` rather than just `ofrak.log` and the name can be specified on the command line.

**Link to Related Issue(s)**
#311 

**Please describe the changes in your request.**
See summary

**Anyone you think should look at this, specifically?**
@whyitfor maybe?

### CODE REVIEW

File: ofrak_core/ofrak/cli/ofrak_cli.py
Comment (rbs-jacob) [inline ofrak_core/ofrak/cli/ofrak_cli.py:132]:
```suggestion
            help="Log file to use; defaults to ofrak<YYYYMMDDhhmmss>.log in a temp directory",
```

**Review by rbs-jacob — CHANGES_REQUESTED** [review]
Change seems fairly straightforward and makes sense to me.

One thing I want to point out (for @whyitfor in particular) is that this changes a top-level API, and may necessitate a major version bump in OFRAK. The argument added to the `OFRAK` initializer is optional, so it shouldn't really break anything, nor should tha block this from being merged. Just wanted to highlight that we should bump semver on the next tagged release.

File: ofrak_core/ofrak/cli/ofrak_cli.py
Comment (rbs-jacob) [inline ofrak_core/ofrak/cli/ofrak_cli.py:128]: If the `DEFAULT_LOG_LEVEL` is a field of the `OFRAK` class, does it make sense for the `DEFAULT_LOG_FILE` path to be as well?

**Review by rbs-jacob — CHANGES_REQUESTED** [review]

File: ofrak_core/ofrak/cli/ofrak_cli.py
Comment (ANogin) [reply ofrak_core/ofrak/cli/ofrak_cli.py:128]: No opinion here - I kept it at top level, where it was, but could be moved into `OFRAK` class, I guess.

Comment (rbs-jacob) [reply ofrak_core/ofrak/cli/ofrak_cli.py:128]: I think it's fine to leave as-is. I'll resolve this conversation and run the tests. Once they pass, this should be good to merge

**Review by rbs-jacob — APPROVED** [review]

**Pull request merged**
